### PR TITLE
Prepend Hermes version CLI output

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #2887

## Description

Add Hermes version to the output of most CLIs.

The command `hermes version` doesn't output duplicate version:

```
hermes version
hermes 1.1.0+240d50e7b
```
<details>
  <summary>Old solution from commit 240d50e7b38be40e65737d45de9ba85f09d7b82e)</summary>

### Example outputs Old solution

`hermes query channels --chain ibc-1`:

```
2022-11-23T08:43:54.041664Z  INFO ThreadId(01) using default configuration from '/Users/luca/.hermes/config.toml'
SUCCESS(v1.1.0+240d50e7b) [
    PortChannelId {
        channel_id: ChannelId(
            "channel-0",
        ),
        port_id: PortId(
            "transfer",
        ),
    },
]
```

`hermes --json query channels --chain ibc-1`:

```
{"timestamp":"2022-11-23T08:45:06.288261Z","level":"INFO","fields":{"message":"using default configuration from '/Users/luca/.hermes/config.toml'"},"threadId":"ThreadId(1)"}
{"result":[{"channel_id":"channel-0","port_id":"transfer"}],"status":"success","version":"1.1.0+240d50e7b"}
```

`hermes query channels --chain ibc-3`:

```
2022-11-23T08:44:46.793190Z  INFO ThreadId(01) using default configuration from '/Users/luca/.hermes/config.toml'
ERROR(v1.1.0+240d50e7b) missing chain config for 'ibc-3' in configuration file
```
</details>

### Example outputs current solution
`hermes query channels --chain ibc-1`:

```
2022-11-23T11:30:25.861000Z  INFO ThreadId(01) using default configuration from '/Users/luca/.hermes/config.toml'
2022-11-23T11:30:25.862253Z  INFO ThreadId(01) Running Hermes v1.1.0+52742f1db
SUCCESS [
    PortChannelId {
        channel_id: ChannelId(
            "channel-0",
        ),
        port_id: PortId(
            "transfer",
        ),
    },
]
```

`hermes --json query channels --chain ibc-1`:

```
{"timestamp":"2022-11-23T11:30:37.823794Z","level":"INFO","fields":{"message":"using default configuration from '/Users/luca/.hermes/config.toml'"},"threadId":"ThreadId(1)"}
{"timestamp":"2022-11-23T11:30:37.825086Z","level":"INFO","fields":{"message":"Running Hermes v1.1.0+52742f1db"},"threadId":"ThreadId(1)"}
{"result":[{"channel_id":"channel-0","port_id":"transfer"}],"status":"success"}
```

`hermes query channels --chain ibc-3`:

```
2022-11-23T11:30:28.496187Z  INFO ThreadId(01) using default configuration from '/Users/luca/.hermes/config.toml'
2022-11-23T11:30:28.497363Z  INFO ThreadId(01) Running Hermes v1.1.0+52742f1db
ERROR missing chain config for 'ibc-3' in configuration file
```
______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Comment by ljoss17** [comment]
> I'd rather that we add an `INFO` message when Hermes starts rather than bolting the version onto the result, which does not make sense for the `start` command.
> 
> We can access the version with `clap::crate_version!()` and output it in one of the lifecycle method in `crates/relayer-cli/src/application.rs`.

Good idea, thank you!
I moved the output to the post-configuration lifecycle. Let me know what you think about this solution.

File: crates/relayer-cli/src/application.rs [hunk 1]
```diff
@@ -132,6 +132,11 @@ impl Application for CliApp {
             }
         };
 
+        tracing::info!(
+            "Using Hermes version: v{}",
+            clap::crate_version!().to_string()
```
Comment (romac) [inline crates/relayer-cli/src/application.rs:137]:
```suggestion
            "Running Hermes v{}",
            clap::crate_version!()
```

**Comment by seanchen1991** [comment]
It seems good to me! What do you think @ancazamfir?

**Review by adizere — APPROVED** [review]
Looks great, thank you Luca!

**Review by ancazamfir — APPROVED** [review]
lgtm! thanks @ljoss17

**Pull request merged**